tests/test_branch_command.rs

```diff
@@ -330,6 +330,128 @@ fn test_branch_forget_deleted_or_nonexistent_branch() {
 
 // TODO: Test `jj branch list` with a remote named `git`
 
+#[test]
+fn test_branch_list_filtered_by_revset() {
+    let test_env = TestEnvironment::default();
+
+    // Initialize remote refs
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "remote", "--git"]);
+    let remote_path = test_env.env_root().join("remote");
+    for branch in ["remote-keep", "remote-delete", "remote-squash"] {
+        test_env.jj_cmd_success(&remote_path, &["new", "root", "-m", branch]);
+        test_env.jj_cmd_success(&remote_path, &["branch", "set", branch]);
+    }
+    test_env.jj_cmd_success(&remote_path, &["new"]);
+    test_env.jj_cmd_success(&remote_path, &["git", "export"]);
+
+    // Initialize local refs
+    let mut remote_git_path = remote_path;
+    remote_git_path.extend([".jj", "repo", "store", "git"]);
```

Comment:
(Optional) If I understand correctly, you're creating a jj repo and only using its local git repo. Perhaps you should instead either create a colocated jj repo or just a git repo? I have such tests in #1700.

Reply:
I just thought using "jj" command is more readable than using git2 API, and I generally avoid colocated repo in tests because it's likely to have some unknown bugs. :)

Reply:
Well, so if you use colocated repos, you're helping debug. :)

Perhaps we need a helper function for creating a jj repo that can push and fetch to/from another repo.

I still don't have any strong opinion here, we can always change it later.